src/public/interfaces.ts

```diff
@@ -394,3 +394,20 @@ export interface OpenConversationRequest {
   */
   onCloseConversation?: (subEntityId: string, conversationId?: string) => void;
 }
+
+export interface IAppLoadFailReason {
+  reason: AppLoadFailReasonTypes;
+  message?: string;
+}
+
+export interface IAppLoadEvent {
+  /**
+   * Indicates that the underlying resource has been created and the settings can be saved.
+   */
+  notifySuccess(): void;
+  /**
+   * Indicates that creation of the underlying resource failed and that the settings cannot be saved.
+   * @param reason Specifies a reason for the failure. If provided, this string is displayed to the user; otherwise a generic error is displayed.
+   */
+  notifyFailure(appLoadFailReason: IAppLoadFailReason): void;
```

Comment:
Can we add an optional string parameter called something like `correlationTag` or `correlationId` that we can use to correlate telemetry between us and our partners? We can sanitize it for PII data and choose to only log this for 1P/2P apps to reduce the risk of PII data getting into our pipeline.

Reply:
Oh, oops... I see that's what the `message` is for. That should be fine.